Fix customize css example

himanshuVerma0404 opened the pull request:
With this PR, I have made the following changes:

1. Added a new power button and circular progress bar control in the front panel.
2. Modified BD to ensure that the newly added circular progress bar also gets the same value as the existing one when we turn on the power button
3. Added rules in the style tag in source panel for both power buttons and circular progress bars to demonstrate Html class attribute use case.
4. Removed controls with [ni-control-id] selector.
5. Remove ‘.section’ CSS rule frpm source panel

There were a few caveats that I took care of:
1. While closing the project only Main.gviweb was opened.
2. Ensured that Application component is not under the hidden dev-target in System Designer.
3. I have not submitted cache files other than .lvprojectcache

I am attaching screenshots for the implementation 

**Main.gviweb**
![customizeexamplewithhtmlclassattr](https://user-images.githubusercontent.com/35248194/51604132-d5108900-1f31-11e9-863a-3b2b0aa97f81.gif)

**Source Panel changes**
![sourcepanel1](https://user-images.githubusercontent.com/35248194/51604194-02f5cd80-1f32-11e9-923a-04a3cc44d40d.png)
![sourcepanel2](https://user-images.githubusercontent.com/35248194/51604195-02f5cd80-1f32-11e9-9b2f-ce12d7ab8859.png)



**Resulting Html**
![powerbuttonhtml](https://user-images.githubusercontent.com/35248194/51523226-357bc980-1e51-11e9-922b-605122cc90f2.png)


**Block Diagram**
![addedcasestructure](https://user-images.githubusercontent.com/35248194/51523380-8e4b6200-1e51-11e9-832b-2f327307498d.png)

himanshuVerma0404 commented:
@msmithNI commented that we might just combine `button-true-bg-yellow` and `button-true-fg-darkgray`  and  into a single `button-true-yellow` rule, but I am thinking of keeping `button-true-fg-darkgray` separate just to show sharing of this class. Also I do not want to repeat `--ni-true-foreground-color: #2b3033` in `button-true-bg-red` too.

himanshuVerma0404 commented:
I have a few doubts:

Do I have to submit the build folder too? If yes then, System Designer -> WebApp.gcomp -> (Right click) Build. The folder generated will be directly submitted?
Do I need to include one more screenshot which will depict the use case of assigning calss in `HTML class attribute` and write rule in source panel accordingly?

himanshuVerma0404 commented:
As per discussion with Christina Rodgers and Marian Cady, I have modified the example statement to:
`This example demonstrates how to customize the user interface of a WebVI using CSS. The user defined CSS is available in the HTML view of this WebVI.`

msmithNI commented on CustomizeWithCss/WebApp.gcomp/Main.gviweb:
```diff
@@ -246,88 +275,108 @@
+				<HtmlBehavior ClassAttributes="progressbar-bg-red" xmlns="http://www.ni.com/PlatformFramework" />
+			</HtmlLVRadialProgressBar>
+			<HtmlLabel Extras="[Dictionary&lt;string, string&gt;]" Height="[float]16" Id="d7b36eef58534e07b6e50d368fbe1dfe" LabelOwner="[UIModel]743c1b230ab6496c94185243448dcca0" Left="[float]418" Text="[string]Start Filling 2" Top="[float]1040" Width="[float]76" xmlns="http://www.ni.com/Controls.LabVIEW.Design" />
+			<HtmlBooleanSwitch BaseName="[string]Power Button" Extras="[Dictionary&lt;string, string&gt;]&quot;checked&quot;=&quot;true&quot;" Height="[float]72" Id="69cbbc0ecbf44173ace1f1cfbe9afb7e" IncludeInCapture="[bool]False" Label="[UIModel]3570b6fb730047c9abe501bee46cfd79" Left="[float]450" MinHeight="[float]20" MinWidth="[float]20" Shape="[SwitchShape]Power" TabIndex="[int]10" Top="[float]1092" Va ...
```
Boolean w/ `checked='true'` in Extras

msmithNI commented on CustomizeWithCss/WebApp.gcomp/Main.gviweb:
```diff
@@ -204,34 +204,63 @@
 		</BlockDiagram>
-		<HtmlFrontPanel Height="[float]1900" Id="14" Left="[float]0" MinHeight="[float]0" MinWidth="[float]0" PanelSizeMode="Fixed" Top="[float]0" UpdatedRootPage="[string]&lt;!DOCTYPE html&gt;&#xA;&lt;html lang=&quot;en&quot;&gt;&#xA;&#xA;    &lt;head&gt;&#xA;&#xA;        &lt;meta charset=&quot;utf-8&quot;&gt;&#xA;        &lt;!---Read only section. Source panel edits to these document properties will  ...
-			<HtmlText BaseName="[string]Text" Extras="[Dictionary&lt;string, string&gt;]&quot;class&quot;=&quot;section&quot;" Foreground="[SMColor]#ffffffff" Height="[float]200" Id="276" Left="[float]0" MinHeight="[float]11" MinWidth="[float]11" SizeMode="[TextModelSizeMode]Fixed" Text="[string]-" TextWrapping="[TextWrapping]Wrap" Top="[float]1473" Width="[float]1000" xmlns="http://www.ni.com/Controls.La ...
+		<HtmlFrontPanel Height="[float]1900" Id="14" Left="[float]0" MinHeight="[float]0" MinWidth="[float]0" PanelSizeMode="Fixed" Top="[float]0" UpdatedRootPage="[string]&lt;!DOCTYPE html&gt;&#xA;&lt;html lang=&quot;en&quot;&gt;&#xA;&#xA;    &lt;head&gt;&#xA;&#xA;        &lt;meta charset=&quot;utf-8&quot;&gt;&#xA;        &lt;!---Read only section. Source panel edits to these document properties will  ...
```
Boolean w/ `checked='true'` in Extras

msmithNI replied:
Boolean w/ `checked='true'` in Extras

msmithNI commented:
Can you open up the Source Panel for the main WebVI, look for `checked='true'` for the 3 booleans, and remove that?
I'm not sure how you ended up in the state of having that attribute added. (I tried opening up the same .gviweb from the unmodified 3.1 examples branch, and resaving, and it didn't happen for me.) It also ends up in the Extras for the .gviweb.

The result is that when you deploy & run this WebVI, the booleans are already on, which is unexpected (doesn't match old behavior, and doesn't match in-IDE behavior). If we did want them on by default, we'd do that via the Default property in the right rail, not source panel changes.

rajsite commented:
> Do I have to submit the build folder too? If yes then, System Designer -> WebApp.gcomp -> (Right click) Build. The folder generated will be directly submitted?

No. Do not change the build output in your PR.

> Do I need to include one more screenshot which will depict the use case of assigning calss in HTML class attribute and write rule in source panel accordingly?

I don't think showing how to set the class in the right rail is necessary but please update the selectors and descriptions to match the front panel and html panel contents

Overall, I want to see the examples on the front panel, the HTML Panel, and the README to look exactly the same. I should be able to scroll down the front panel then open the html source panel and scroll through the style section and they look exactly the same. Finally scrolling through the README the examples should also look the same.

To help with that, I noticed the following inconsistencies, but ultimately that is what I'm looking for to review this example.


Front panel examples to remove (stuff not in html panel):

The following selector is not used anymore:
```
body {
    background-color: white;
}
```

The following should be moved from next to the slider multiplier style to the top section:
```
.background-white {
    background-color: white;
}
```

This comment seems wrong now:
Define a style for 'sections' with a white background color. The white sections on this page are Text elements, with class="section" set on them from the HTML view.

It looks like the white sections have the class "background-white" not the class "section"

The following is not used anymore:
```
ni-cartesian-axis[ni-control-id] {
    color: white;
}
```

HTML panel examples to remove:

It looks like these two selectors are not needed anymore because they were replaced with the classes example:
```
jqx-power-button {
    --ni-true-background: yellow;
    --ni-true-foreground-color: #2b3033;
}
            
jqx-circular-progress-bar {
    --ni-fill-background: yellow;
}
```

himanshuVerma0404 commented:
I made changes as per suggestion.
1. Removed `checked = true` from boolean switch and power button controls
2. Removed unused selectors.
3. Modified Readme.md file to be consistent with front panel and source panel changes.

himanshuVerma0404 commented:
ReadMe.md file has a screenshot.gif whose path is same as Screenshot.gif in the outer folder. I guess changing Screenshot.gif (outer one) will automatically update the one in ReadMe.md file

msmithNI commented:
I verified that the checked=true is gone from the booleans.
Part of Milan's feedback was that the ordering of the CSS classes as specified in the Source Panel didn't match the order of the same CSS text on the front panel, so I pushed an update fixing that (and making the font of one CSS text item Courier New like the others; and resized the numerics at the bottom a bit since they were misaligned (too large for their font size)).
